### PR TITLE
Add logic to check if luci try builders are enabled

### PR DESCRIPTION
**Pull request by keyonghan**
This is part of https://github.com/flutter/flutter/issues/63437

### CODE REVIEW

**Comment by keyonghan** [comment]
> Do we need to update the try json files in the different repos first?

We can do this in parallel. `where((Map<String, dynamic> element) => (element['enabled'] as bool) ?? true)` considers the null case, replacing with default `true`.

File: app_dart/lib/src/foundation/utils.dart [hunk 1]
```diff
@@ -95,7 +97,10 @@ Future<List<Map<String, dynamic>>> getRepoBuilders(HttpClientProvider branchHttp
   Map<String, dynamic> builderMap;
   builderMap = json.decode(builderContent) as Map<String, dynamic>;
   final List<dynamic> builderList = builderMap['builders'] as List<dynamic>;
-  return builderList.map((dynamic builder) => builder as Map<String, dynamic>).toList();
+  return builderList
+      .map((dynamic builder) => builder as Map<String, dynamic>)
+      .where((Map<String, dynamic> element) => (element['enabled'] as bool) ?? true)
+      .toList();
```
Comment (godofredoc) [inline app_dart/lib/src/foundation/utils.dart:103]: Do we need to add the property to the model class?

File: app_dart/test/foundation/utils_test.dart [hunk 2]
```diff
@@ -118,11 +120,11 @@ void main() {
         lucBuilderHttpClient = FakeHttpClient();
         log = FakeLogging();
       });
-      test('returns luci builders', () async {
+      test('returns enabled luci builders', () async {
         lucBuilderHttpClient.request.response.body = luciBuilders;
         final List<Map<String, dynamic>> builders =
             await getRepoBuilders(() => lucBuilderHttpClient, log, (int attempt) => Duration.zero, 'try', 'cocoon');
-        expect(builders.length, 2);
+        expect(builders.length, 1);
```
Comment (godofredoc) [inline app_dart/test/foundation/utils_test.dart:127]: Can we also add a test for the validation of the model class?

File: app_dart/lib/src/foundation/utils.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/lib/src/foundation/utils.dart:103]: The model class is for prod builders usage only. We don't need `enabled` flag for prod builders, right?

Comment (godofredoc) [reply app_dart/lib/src/foundation/utils.dart:103]: > The model class is for prod builders usage only. We don't need `enabled` flag for prod builders, right?

Good point I though we were using the model class for try too.

**Pull request merged**